### PR TITLE
feat(add 2.19 doc to opentronsai): upgrading documentation 

### PR DESCRIPTION
**Pull request by Elyorcv**
Closes closes [AUTH-630](https://opentrons.atlassian.net/browse/AUTH-630)

# Overview
Currently OpentronsAI uses 2.15. This PR upgrades to 2.19.

## Test Plan and Hands on Testing
- Run server: 
```
cd opentrons-ai-server  
make local-run
```

- Run client:
```
cd opentrons-ai-client
make dev
```

Play with UI, you might type as follows:
image

## Changelog
- Pushed indexed Python protocol API 2.19
- Pushed file that shows how to create index file
- Updated UI such that one can run the app locally

## Risk assessment
Low

[AUTH-630]: https://opentrons.atlassian.net/browse/AUTH-630?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: opentrons-ai-client/src/main.tsx [hunk 1]
```diff
@@ -8,21 +8,39 @@ import { i18n } from './i18n'
...
     <React.StrictMode>
       <Auth0Provider
-        domain={AUTH0_DOMAIN}
-        clientId={
-          process.env.NODE_ENV === 'production'
-            ? PROD_AUTH0_CLIENT_ID
-            : STAGING_AUTH0_CLIENT_ID
-        }
+        clientId={clientId}
+        domain={getDomain()}
```
Comment (koji) [inline opentrons-ai-client/src/main.tsx]: for consistency

```ts
const clientId = getClientId()
const domain = getDomain()
  ReactDOM.createRoot(rootElement).render(
    <React.StrictMode>
      <Auth0Provider
        clientId={clientId}
        domain={domain}`
```

File: opentrons-ai-client/src/resources/hooks/useGetAccessToken.ts [hunk 2]
```diff
@@ -1,16 +1,35 @@
...
+      default:
+        console.error(
+          'Error: NODE_ENV variable is not valid:',
+          process.env.NODE_ENV
+        )
+        return STAGING_AUTH0_AUDIENCE
+    }
+  }
+
+  const auth0Audience = getAuth0Audience()
```
Comment (koji) [inline opentrons-ai-client/src/resources/hooks/useGetAccessToken.ts]:
```suggestion
  const auth0Audience = (): string => {
    switch (process.env.NODE_ENV) {
      case 'production':
        return PROD_AUTH0_AUDIENCE
      case 'staging':
        return STAGING_AUTH0_AUDIENCE
      case 'development':
        return LOCAL_AUTH0_AUDIENCE
      default:
        console.error(
          'Error: NODE_ENV variable is not valid:',
          process.env.NODE_ENV
        )
        return STAGING_AUTH0_AUDIENCE
    }
  }

```

File: opentrons-ai-server/api/utils/create_index.py [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+
+from llama_index.core import Settings as llamasettings
+from llama_index.core import SimpleDirectoryReader, StorageContext, VectorStoreIndex, load_index_from_storage
+from llama_index.embeddings.openai import OpenAIEmbedding
+
+ROOT_PATH: Path = Path(Path(__file__)).parent.parent.parent
+llamasettings.embed_model = OpenAIEmbedding(model_name="text-embedding-3-large", api_key=os.environ["OPENAI_API_KEY"])
+
+
+def create_index(data_path: str, data_file: str, index_name: str) -> Any:
```
Comment (koji) [inline opentrons-ai-server/api/utils/create_index.py]: we should avoid using `Any` as much as possible.

Comment (Elyorcv) [reply opentrons-ai-server/api/utils/create_index.py]: @koji have a look thanks!

File: opentrons-ai-client/src/resources/hooks/useGetAccessToken.ts [hunk 4]
```diff
@@ -1,22 +1,39 @@
 import { useAuth0 } from '@auth0/auth0-react'
-import { PROD_AUTH0_AUDIENCE, STAGING_AUTH0_AUDIENCE } from '../constants'
+import {
+  PROD_AUTH0_AUDIENCE,
+  STAGING_AUTH0_AUDIENCE,
+  LOCAL_AUTH0_AUDIENCE,
```
Comment (koji) [inline opentrons-ai-client/src/resources/hooks/useGetAccessToken.ts]: we use VSCode to sort import statements `shift + cmd + p`


https://github.com/user-attachments/assets/089a3fd4-d1a1-4e2c-8312-8775988abe17



```suggestion
  LOCAL_AUTH0_AUDIENCE,
  PROD_AUTH0_AUDIENCE,
  STAGING_AUTH0_AUDIENCE,
```

**Review by koji — APPROVED** [review]
lgtm

File: opentrons-ai-client/src/resources/hooks/useGetAccessToken.ts (hunk 4 above)
Comment (Elyorcv) [reply opentrons-ai-client/src/resources/hooks/useGetAccessToken.ts]: Many thanks Koji!

**Pull request merged**